### PR TITLE
streamline logging

### PR DESCRIPTION
**Pull request by vroldanbet**
Part of https://github.com/authzed/spicedb/issues/753
Built on top of https://github.com/jzelinskie/cobrautil/pull/19

# What

Adjustments to homogenize and streamline how logging is used in SpiceDB.

# How

Not all logging messages have been revisited. That will be tackled in a follow up. This PR sets the foundation.

- migrates to cobrautil v2
- makes it possible to stop relying on zerolog global logger, and thus be able to have our own default logger (disabled) 
- configures logger to be non-blocking
- configures same logger for OTel library
- logging is now disabled by default in our tests (no more need to do `zerolog.SetGlobalLevel(zerolog.Disabled)` in each test)
- `zerolog.Ctx()` is used in the codebase frequently, but because [we don't set the default logger for contet-based logging](https://github.com/rs/zerolog/blob/master/globals.go#L108-L110). We are now setting `zerolog.DefaultContextLogger` whenever our own global logger is set.
- [add missing Close on http downloader server and log error](https://github.com/authzed/spicedb/pull/844/files#diff-cd758728458ab57fbaf0644ea41b2c62aeb417a789098b6cf3bd20316b95030cR110-R112)

The changeset is relatively simple, it's just that lots of imports had to be adjusted. Files to look into are:
- https://github.com/authzed/spicedb/pull/844/files#diff-a47342e8d77d583a1cdd9b90f604b523ddfc98b1135231779f8b8d3de2de32cc
- https://github.com/authzed/spicedb/pull/844/files#diff-bd140be825bc795319d8e428c19dd9c6c726167777d27b346718f52b801a368b
- https://github.com/authzed/spicedb/pull/844/files#diff-df0635588ca5d6a577487faacc185c3838a3fe2b321660c36372541797c54c74

### Example startup at `INFO` level

image

### Example startup at `DEBUG` level

image

### CODE REVIEW

File: internal/logging/logger.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"fmt"
+
+	"github.com/rs/zerolog"
+)
+
+var Logger zerolog.Logger
+
+func init() {
+	nop := zerolog.Nop()
+	SetGlobalLogger(nop)
```
Comment (jzelinskie) [inline internal/logging/logger.go:14]: why not `SetGlobalLogger(zerolog.Nop())`?

File: internal/logging/logger.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	return Logger.Fatal()
+}
+
+func WithLevel(level zerolog.Level) *zerolog.Event {
+	return Logger.WithLevel(level)
+}
+
+func Log() *zerolog.Event {
+	return Logger.Log()
+}
```
Comment (jzelinskie) [inline internal/logging/logger.go:60]: You can actually make these all one line each and gofmt won't complain:

```go
func With() zerolog.Context { return Logger.With() }
func Err(err error) *zerolog.Event { return Logger.Err(err) }
func Trace() *zerolog.Event { return Logger.Trace() }
func Debug() *zerolog.Event { return Logger.Debug() }
func Info() *zerolog.Event { return Logger.Info() }
func Warn() *zerolog.Event { return Logger.Warn() }
func Error() *zerolog.Event { return Logger.Error() }
func Fatal() *zerolog.Event { return Logger.Fatal() }
func WithLevel(level zerolog.Level) *zerolog.Event { return Logger.WithLevel(level) }
func Log() *zerolog.Event { return Logger.Log() }
```

File: pkg/cmd/root.go [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
 	"github.com/authzed/spicedb/pkg/cmd/server"
 	"github.com/authzed/spicedb/pkg/releases"
 )
 
 func RegisterRootFlags(cmd *cobra.Command) {
-	cobrautil.RegisterZeroLogFlags(cmd.PersistentFlags(), "log")
-	cobrautil.RegisterOpenTelemetryFlags(cmd.PersistentFlags(), "otel", cmd.Use)
+	cobrazerolog.New().RegisterFlags(cmd.PersistentFlags())
+	cobraotel.New("spicedb").RegisterFlags(cmd.PersistentFlags())
```
Comment (jzelinskie) [inline pkg/cmd/root.go:14]: I think we might prefer to keep this cmd.Use so that the binary name is not hard coded

**Review by jzelinskie — CHANGES_REQUESTED** [review]
Only minor nitpicks

File: pkg/cmd/root.go (hunk 3 above)
Comment (vroldanbet) [reply pkg/cmd/root.go:14]: ah! good catch!

File: internal/logging/logger.go (hunk 1 above)
Comment (vroldanbet) [reply internal/logging/logger.go:14]: leftover from it being a pointer. Instafix!

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**